tests/api2/pool_dataset_encryption.py

```diff
@@ -0,0 +1,870 @@
+        }
+    }
+    results = POST('/pool/dataset/lock', payload)
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_38_verify_passphrase_encrypted_root_is_locked():
+    payload = {
+        'id': dataset
+    }
+    results = POST('/pool/dataset/encryption_summary/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+    job_status_result = job_status['results']['result']
+    for dictionary in job_status_result:
+        if dictionary['name'] == dataset:
+            assert dictionary['locked'] is True, str(job_status_result)
+            break
+    else:
+        assert False, str(job_status_result)
+
+
+@skip_for_ha
+def test_39_unlock_passphrase_encrypted_datasets():
+    payload = {
+        'id': dataset,
+        'unlock_options': {
+            'recursive': True,
+            'datasets': [
+                {
+                    'name': dataset,
+                    'passphrase': 'my_passphrase'
+                }
+            ]
+        }
+    }
+    results = POST('/pool/dataset/unlock/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+
+
+@skip_for_ha
+def test_40_verify_passphrase_encrypted_root_is_unlocked():
+    payload = {
+        'id': dataset
+    }
+    results = POST('/pool/dataset/encryption_summary/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+    job_status_result = job_status['results']['result']
+    for dictionary in job_status_result:
+        if dictionary['name'] == dataset:
+            assert dictionary['locked'] is False, str(job_status_result)
+            break
+    else:
+        assert False, str(job_status_result)
+
+
+@skip_for_ha
+def test_41_delete_encrypted_dataset():
+    results = DELETE(f'/pool/dataset/id/{dataset_url}/')
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_42_create_an_not_encrypted_dataset_on_a_key_encrypted_pool():
+    payload = {
+        'name': dataset,
+        'encryption': False,
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_43_delete_encrypted_dataset():
+    results = DELETE(f'/pool/dataset/id/{dataset_url}/')
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_44_create_an_dataset_with_inherit_encryption_from_the_key_encrypted_pool():
+    payload = {
+        'name': dataset,
+        'inherit_encryption': True
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 200, results.text
```

Comment:
Please verify that it is indeed encrypted and it's encryption root is the parent ( which is root dataset in this case )